operator/src/test/java/org/keycloak/operator/testsuite/integration/RealmImportTest.java

```diff
@@ -99,6 +110,32 @@ public void testWorkingRealmImport() {
         K8sUtils.set(k8sclient, getClass().getResourceAsStream("/example-realm.yaml"));
 
         // Assert
+        assertWorkingRealmImport(kc, true);
+    }
+
+    @Test
+    public void testWorkingRealmImportNoReplacement() {
+        // Arrange
+        var kc = getTestKeycloakDeployment(false);
+        kc.getSpec().setImage(null); // checks the job args for the base, not custom image
+        kc.getSpec().setImagePullSecrets(Arrays.asList(new LocalObjectReferenceBuilder().withName("my-empty-secret").build()));
+        deployKeycloak(k8sclient, kc, false);
+
+        // Act
+        K8sUtils.set(k8sclient, getClass().getResourceAsStream("/example-realm.yaml"), obj -> {
```

Comment:
@stustison the tests are failing because this manipulation logic does not seem to be removing the placeholders. Also having the placeholders in the base yaml is causing another testWorkingRealmImportWithCustomImage to fail because the secret is not deployed - maybe this should be reversed? Remove the placeholeders from the yaml and add the placeholders / secret in a specific test.

Reply:
Ok I'll take a look and switch this around then. Thanks